x-pack/elastic-agent/pkg/agent/transpiler/rules.go

```diff
@@ -429,6 +448,111 @@ func InjectIndex(indexType string) *InjectIndexRule {
 	}
 }
 
+// InjectStreamProcessorRule expect target to be a collection of fields including
+// _meta_index map with dataset, index and namespace keys defined.
+// if any key is missing this rule fails.
+type InjectStreamProcessorRule struct {
+	Target string
+}
+
+// Apply injects processor into input.
+func (r *InjectStreamProcessorRule) Apply(ast *AST) error {
+	const defaultNamespace = "default"
+	const defaultDataset = "generic"
+
+	inputsNode, found := Lookup(ast, r.Target)
+	if !found {
+		return nil
+	}
+
+	inputsList, ok := inputsNode.Value().(*List)
+	if !ok {
+		return nil
+	}
+
+	for i := range inputsList.value {
+		inputNode := inputsList.value[i]
+		metaNode, ok := inputNode.Find(metaIndexKey)
```

Comment:
this couples this rule with index rule, DECOUPLE

Reply:
Just ping me when you want me to take another look after you do your modification.

Reply:
i think it's ready for review, i decoupled for a cost of additional tree traversal and i introduce on conflict merge logic